BillTrack.Api/Features/Departments/Endpoints/GetAllDepartments.cs

```diff
@@ -0,0 +1,35 @@
+using AutoMapper.QueryableExtensions;
+using BillTrack.Core.Contracts.Department;
+using BillTrack.Core.Interfaces.Services;
+using BillTrack.Domain.Entities;
+using FastEndpoints;
+using IMapper = AutoMapper.IMapper;
+
+namespace BillTrack.Api.Features.Departments.Endpoints;
+
+public class GetAllDepartments : EndpointWithoutRequest<IQueryable<DepartmentResponse>>
+{
+    private readonly IWebApiService _webApiService;
+    private readonly IMapper _mapper;
+    
+    public GetAllDepartments(IWebApiService webApiService, IMapper mapper)
+    {
+        _webApiService = webApiService;
+        _mapper = mapper;
+    }
+    
+    public override void Configure()
+    {
+        Get("departments");
+        AllowAnonymous();
+    }
+    
+    public override async Task HandleAsync(CancellationToken c)
+    {
+        var entities = _webApiService.GetAll<Department>();
```

Comment:
`GetAll()` without parameters is a bad practice. If the table could contain millions of records then all should be returned.
Usually, paging parameters are provided. There should be default values to have a fallback.
Please, add parameters to this endpoint, setup default values. And pass them to the _webApiService's method. The update could be done in the scope of this task or further one.

Reply:
fixed